src/React.Core/Babel.cs

```diff
@@ -350,12 +350,36 @@ string filename
 		)
 		{
 			var outputPath = GetOutputPath(filename);
-			var sourceMapPath = GetSourceMapOutputPath(filename);
 			var contents = _fileSystem.ReadAsString(filename);
-			var result = TransformWithHeader(filename, contents, null);
-			_fileSystem.WriteAsString(outputPath, result.Code);
-			_fileSystem.WriteAsString(sourceMapPath, result.SourceMap == null ? string.Empty : result.SourceMap.ToJson());
-			return outputPath;
+            if (MustTranspileFile(contents, outputPath))
```

Comment:
The reason I didn't do that in the first place is that NOTs (!) in C# is rather subtle and easy to miss. Renaming the MustTranspileFile to e.g. CacheIsValid would allow me to do that. Good suggestion.

It would give two ```return outputPath;``` statements, though.